projects/vdk-core/src/vdk/internal/builtin_plugins/termination_message/writer.py

```diff
@@ -25,13 +30,25 @@ def vdk_configure(self, config_builder: ConfigurationBuilder):
 
     @hookimpl
     def vdk_exit(self, context: CoreContext, exit_code: int):
-        self._write_termination_message(
+        self.write_termination_message(
             errors.get_blamee_overall(),  # TODO: get this from context
             errors.get_blamee_overall_user_error(),  # TODO: get this from context
             context.configuration,
         )
 
-    def _write_termination_message(self, error_overall, user_error, configuration):
+    # Returns the version of the package named dist_name
+    @staticmethod
+    def version(
+        dist_name="vdk-core",
```

Comment:
This would record the version of vdk-core. 

I think we need the version of the main distribution of vdk that aggregates different plugins and vdk-core (e.g quickstart-vdk, company-x-vdk). 

From user point of view, this is the main version they'd be using and they know. Also from the vdk distribution version we can see the version of the plugins potentially.

Reply:
You pray to the gods of ancient Rome and Greece and then fast for 30 days and the path would show for you. 

Alternatively, we can re-use PACKAGE_NAME - https://github.com/vmware/versatile-data-kit/blob/4cdbe3b0cfedb8ddc675bec27aee4a88389ce16d/projects/vdk-core/src/vdk/internal/builtin_plugins/version/new_version_check_plugin.py#L30 - whose idea is to set the correct main python package (or distribution) name to use to check for the new version. 

But we'd need to. fix the description and possibly refactor a bit (so it's not used only for new version check)

Reply:
I agree, but let's do it iteratively in a separate PR.